### PR TITLE
Creating an audiobook with Pyttxs3, PyPDF3 and Tkinter

### LINKED ISSUE #2499: Creating an Audiobook with PyPDF3 and Pyttsx3
### Proposed title of the article
Creating an Audiobook with PyPDF3 and Pyttsx3 (Python libraries)

### Introduction paragraph (2-3 paragraphs):
There's has never been a time in the history of humankind with so much knowledge to consume. This has caused a distracted world where what you read is what makes you. This, in turn, caused time deficiency among readers which has led to the creation of audiobooks. Whether working, driving or eating, you can consume a book.

But as a developer, you need not buy one when you can actually create one for free.

In this tutorial, we shall look at PyPDF3 and Pyttsx3 (Python libraries) and how to use them to create your own audiobook.

### Key takeaways:
* What isPyPDF3?
* What is Pyttsx3?
* Using both PyPDF3 and Pyttsx3 to create, use and save an audiobook in a .mp3 format.

### References:
* [Pyttsx3](https://pypi.org/project/pyttsx3/)
* [PyPDF3](https://pypi.org/project/PyPDF3/)

### PR DESCRIPTION
**Pull request by FREDERICO23**
### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

This fixes  #2499

### CODE REVIEW

**Comment by hectorkambow** [comment]
@FREDERICO23 
Good afternoon and thank you for submitting your article for publishing. It has been entered into our queue and should be reviewed (and published pending any changes) as soon as a peer reviewer is finished with the ones in the queue before it. 🚀

Also FYI - I see you have made a few commits after creating the PR - please only create a PR when the article is fully polished and ready to be reviewed.

Otherwise you can make all those changes and commits on your forked repo 👍

Just an FYI to avoid potential closure of future articles. Thank you.

**Comment by FREDERICO23** [comment]
Thanks for the FYI

**Comment by ahmadmardeni1** [comment]
If you have time, can you review this PR? @Bayler

**Comment by Bayler** [comment]
> If you have time, can you review this PR? @Bayler

Sure thing @ahmadmardeni1

**Comment by hectorkambow** [comment]
@FREDERICO23 
There are TWO Articles here - the PR should only have one article?? 

Please verify the PR is submitted correctly - each article should be a new branch on your forked repo.

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by Bayler** [comment]
> @FREDERICO23
> There are TWO Articles here - the PR should only have one article??
> 
> Please verify the PR is submitted correctly - each article should be a new branch on your forked repo.
> 
> cc @Bayler

@FREDERICO23 Please make the necessary changes so that I can continue with the review

**Comment by FREDERICO23** [comment]
@Bayler I'm into it. Please give me a few hours to correct it

**Comment by FREDERICO23** [comment]
@Bayler It is done you can now continue with the review

**Comment by Bayler** [comment]
@FREDERICO23 This is an interesting read. Tt is also easy to follow and read. Good job! 

Just a few questions:

1. Do you have a link to a working code example on replit for this project?
2. Are you using a windows machine or Linux machine? It's not clear.
3. Is this our first article contribution on section?

File: content/articles/creating an audio book using PDF3, Tkinter and Python Text to Speech(Pyttsx3)/index.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+1. Knowledge of Python.
+2. Python 3 installed in their computer.
+3. Python text to speech library version 3.
+4. Python PDF version 3.
+5. Understanding of [Tkinter](https://docs.python.org/3/library/tk.html).
+
+### What is PyPDF3?
+
+From PyPDF3's official documentation, it is a Pure-Python library built as a PDF toolkit. It was built to help with the following:
+
```
Comment (Bayler) [inline content/articles/creating an audio book using PDF3, Tkinter and Python Text to Speech(Pyttsx3)/index.md:22]: Please add the links to PyPDF3 official documentation

File: content/articles/creating an audio book using PDF3, Tkinter and Python Text to Speech(Pyttsx3)/index.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+> It is important to note that since this library is built from a file page perspective, it has problems manipulating pdf files that are not page numbered.
+
+### Application
+
+- It can be used as a tool for websites that manage or manipulate PDFs.
+
+### What is Pyttsx3?
+
+From the official documentation, Python Text to Speech version 3 (`pyttsx3`) is a text-to-speech conversion library in Python. Unlike alternative libraries, it works offline, and is compatible with both Python 2 and 3.
```
Comment (Bayler) [inline content/articles/creating an audio book using PDF3, Tkinter and Python Text to Speech(Pyttsx3)/index.md:38]: Please add the links to Pyttsx3 official documentation

**Comment by FREDERICO23** [comment]
> @FREDERICO23 This is an interesting read. It is also easy to follow and read. Good job!
> **Thank you**
> Just a few questions/comments:
> 
> 1. Do you have a link to a working code example on [Replit](https://replit.com/) for this project? *No*
> 2. Are you using a windows machine or Linux machine? It's not clear.  *I'm using Windows*
> 3. Is this your first article contribution on section? Or your second? *My second*
> 4. Make sure your article has 750 words and above . *Let me check the word count.*

**Comment by Bayler** [comment]
> > @FREDERICO23 This is an interesting read. It is also easy to follow and read. Good job!
> > **Thank you**
> > Just a few questions/comments:
> > 
> > 1. Do you have a link to a working code example on [Replit](https://replit.com/) for this project? _No_
> > 2. Are you using a windows machine or Linux machine? It's not clear.  _I'm using Windows_
> > 3. Is this your first article contribution on section? Or your second? _My second_
> > 4. Make sure your article has 750 words and above . _Let me check the word count._

@FREDERICO23 Please clearly state inside your article on the prerequisite section that you're using a Windows OS machine. It can be confusing to the reader as they might be using a Linux machine, so it is important to clearly point that out. Also, it is important to include your working code on replit then include the link inside your article. Readers can test your code on replit.

**Comment by FREDERICO23** [comment]
Okay let me work on it.

**Comment by FREDERICO23** [comment]
Please help me test the code from your end.

**Comment by Bayler** [comment]
@FREDERICO23 I've noted a problem with your article folder structure which I had missed earlier. Please read our [How to upload an article for Section’s Engineering Education Program](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/UPLOAD_INSTRUCTIONS.md).

For example, your url within the articles folder (inside the content folder), is off. It should be something like this:

**creating-an-audiobook-with-pyttxs3-pypdf3-and-tkinter/index.md**

Be sure to:

1. Use all lowercase.
2. Use dashes in place of spaces.
3. Do not include any special characters.
4. Try to keep it under 40 characters.
5. Avoid using periods (.) or commas (,) or (_ ) underscores

**Comment by FREDERICO23** [comment]
@Bayler I corrected it. I don't know how I missed that.

**Comment by Bayler** [comment]
> @Bayler I corrected it. I don't know how I missed that.

@FREDERICO23  Your article contains two articles. You've made the revisions I asked but again re-introduced the article we requested you deleted. Please remove them.


> @FREDERICO23
> There are TWO Articles here - the PR should only have one article??
> 
> Please verify the PR is submitted correctly - each article should be a new branch on your forked repo.
> 
> cc @Bayler

@FREDERICO23  Your article contains two articles. You've made the revisions I asked but again re-introduced the article we requested you deleted in the quoted reply. Please remove them.

**Comment by Bayler** [comment]
@FREDERICO23 have you removed the extra files? I should be finishing up with this review.

**Comment by FREDERICO23** [comment]
@Bayler Yes I have removed the extra files you can proceed with the review

**Comment by Bayler** [comment]
> @Bayler Yes I have removed the extra files you can proceed with the review

@FREDERICO23 Are you sure? I can still see them. They should disappear from the "Files changed" section

**Comment by FREDERICO23** [comment]
I just deleted them physically, git rebase proved to mess my work more so I opted to just delete them which leaves a footprint which git rebase would not have left

**Comment by FREDERICO23** [comment]
@Bayler Kindly check if it affects the PR anyway

**Comment by Bayler** [comment]
> @Bayler Kindly check if it affects the PR anyway

@ahmadmardeni1 please help me address this issue if it affects the PR in any way. It is the issue concerning two articles in one PR which has been deleted

**Comment by ahmadmardeni1** [comment]
I'm not sure why there are 2 index.md & hero images. And I tried to delete them but they are still there.

Please open another PR and close this one, just to make sure everything is good, and it does not overlap when we merge it.

@ me when you are done with the review and I will do the final review & publish it ASAP. @Bayler @FREDERICO23

**Comment by FREDERICO23** [comment]
@Bayler @ahmadmardeni1 I'm closing this PR and opening one in a few minutes. Pinging you after it is done.

**Pull request closed without merging**